client/lib/username/index.js

```diff
@@ -1,117 +0,0 @@
-/**
- * External dependencies
- */
-import i18n from 'i18n-calypso';
-
-/**
- * Internal dependencies
- */
-import Emitter from 'calypso/lib/mixins/emitter';
-import wpcom from 'calypso/lib/wp';
-import user from 'calypso/lib/user';
-
-/**
- * Initialize Username with defaults
- */
-function Username() {
-	if ( ! ( this instanceof Username ) ) {
-		return new Username();
-	}
-
-	this.validation = false;
-	this.hasUsernameSite = false;
-}
-
-Emitter( Username.prototype );
-
-Username.prototype.validate = function ( username ) {
-	if ( username !== user().get().username ) {
-		if ( username.length < 4 ) {
-			this.validation = {
-				error: 'invalid_input',
-				message: i18n.translate( 'Usernames must be at least 4 characters.' ),
-			};
-
-			this.emit( 'change' );
-		} else if ( ! /^[a-z0-9]+$/.test( username ) ) {
-			this.validation = {
-				error: 'invalid_input',
-				message: i18n.translate(
-					'Usernames can only contain lowercase letters (a-z) and numbers.'
-				),
-			};
-
-			this.emit( 'change' );
-		} else {
-			wpcom
-				.undocumented()
-				.me()
-				.validateUsername(
-					username,
-					function ( error, data ) {
-						if ( error ) {
-							this.validation = error;
-						} else {
-							this.validation = data;
-							this.validation.validatedUsername = username;
-						}
-
-						this.emit( 'change' );
-					}.bind( this )
-				);
-		}
-	} else {
-		this.validation = false;
-		this.emit( 'change' );
-	}
-};
-
-Username.prototype.change = function ( username, action, callback ) {
-	wpcom
-		.undocumented()
-		.me()
-		.changeUsername(
-			username,
-			action,
-			function ( error, data ) {
-				if ( error ) {
-					this.validation = error;
-					this.emit( 'change' );
-				} else {
-					user().fetch();
```

Comment:
If you look closely you'll notice that I removed this line of code in my refactoring. If I'm not mistaken it's not necessary to keep it. We reload the whole page after the username change is successful so this won't have any effect. We also don't wait for it to resolve in the old code before the page reload. I may be wrong though.